src/bp-members/bp-members-functions.php

```diff
@@ -4853,15 +4853,18 @@ function bb_is_online_user( $user_id, $expiry = false ) {
 	}
 
 	if ( true === $expiry ) {
-		$timeframe = apply_filters( 'bb_is_online_user_expiry', 300 ); // Default 300 seconds.
+		$interval_time = bb_presence_interval();
+		$timeframe     = $interval_time + 5;
```

Comment:
add the 5 sec into a variable called extra_time_span. It should be 20sec by default, considering the longest ajax call to be 20 sec.

this variable can be changeable using a filter also.